### PR TITLE
feat(linter): no irregular whitespace

### PR DESCRIPTION
**Pull request by DeividAlmeida**
it was builded a loop that iterates in trivias.irregular-whitespace and return the warning if contain some specific sequence of chars.

### CODE REVIEW

**Comment by DeividAlmeida** [comment]
Do you have any suggestion to improve the performance?

**Comment by DeividAlmeida** [comment]
do you have any suggestion to improve the performance?

File: crates/oxc_linter/src/rules/eslint/no_irregular_whitespace.rs [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+impl Rule for NoIrregularWhitespace {
+    fn run_once(&self, ctx: &LintContext) {
+        let irregular_whitespaces = ctx.semantic().trivias().irregular_whitespaces().to_owned();
+        let source = ctx.source_text();
+
+        for irregular_whitespace in irregular_whitespaces {
+            let char = ctx
+                .source_text()
+                .get(irregular_whitespace.start as usize..irregular_whitespace.end as usize)
+                .unwrap();
```
Comment (Dunqing) [inline crates/oxc_linter/src/rules/eslint/no_irregular_whitespace.rs:45]:
```suggestion
            let char = irregular_whitespace.source_text(ctx.source_text())
```

**Review by Dunqing — CHANGES_REQUESTED** [review]
I don't think we need to check irregular_whitespaces again. we can loop over `irregular_whitespaces` and just call `ctx.diagnostic(NoIrregularWhitespaceDiagnostic(irregular_ whitespace))` and that's it!

File: crates/oxc_linter/src/rules/eslint/no_irregular_whitespace.rs [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+    ///     return  42;
+    /// }
+    /// ```
+    NoIrregularWhitespace,
+    correctness
+);
+
+impl Rule for NoIrregularWhitespace {
+    fn run_once(&self, ctx: &LintContext) {
+        let irregular_whitespaces = ctx.semantic().trivias().irregular_whitespaces().to_owned();
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/no_irregular_whitespace.rs:38]: `to_owned` is a clone here I think, which is unecessary.

**Comment by DeividAlmeida** [comment]
> I don't think we need to check irregular_whitespaces again. we can loop over `irregular_whitespaces` and just call `ctx.diagnostic(NoIrregularWhitespaceDiagnostic(irregular_ whitespace))` and that's it!

if we don't check it'll not pass and all tests, but i have some doubts about the tests relevance, can i proceed without checks?

**Comment by Dunqing** [comment]
> > I don't think we need to check irregular_whitespaces again. we can loop over `irregular_whitespaces` and just call `ctx.diagnostic(NoIrregularWhitespaceDiagnostic(irregular_ whitespace))` and that's it!
> 
> if we don't check it'll not pass and all tests, but i have some doubts about the tests relevance, can i proceed without checks?

I think we can comment out the failed test first. And then figure out how to fix it.

**Comment by Boshen** [comment]
> I think we can comment out the failed test first. And then figure out how to fix it.

I Agree. It seems like most of the tests can be fixed by fixing the lexer, it's not saving all the irregular whitespaces.

**Review by Dunqing — APPROVED** [review]

**Comment by Boshen** [comment]
Thank you!

(I removed the redundant `.clone()`.)

**Pull request merged**